# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1684583469 -7200
# Parent  25f9d021aa18e4c80154be5de9ced2343e3c702f
Misc: Move unified toolbar after menu. Put minimise/fullscreen/close buttons into menubar and hide after unified toolbar when necessary.

diff --git a/mail/base/content/messenger.xhtml b/mail/base/content/messenger.xhtml
--- a/mail/base/content/messenger.xhtml
+++ b/mail/base/content/messenger.xhtml
@@ -460,17 +460,16 @@
   GTK needs to draw behind the lightweight theme toolbox backgrounds, thus the
   extra box. Also this box allows a negative margin-top to slide the toolbox off
   screen in fullscreen layout.
 -->
 <box id="navigation-toolbox-background">
   <toolbox id="navigation-toolbox" flex="1" labelalign="end" defaultlabelalign="end">
 
     <vbox id="titlebar">
-      <html:unified-toolbar></html:unified-toolbar>
       <!-- Menu -->
       <toolbar is="customizable-toolbar" id="toolbar-menubar"
                class="chromeclass-menubar themeable-full"
                type="menubar"
                customizable="true"
                toolboxid="mail-toolbox"
 #ifdef XP_MACOSX
                defaultset="menubar-items"
@@ -486,17 +485,19 @@
                mode="icons"
                insertbefore="tabs-toolbar"
                prependmenuitem="true">
         <toolbaritem id="menubar-items" align="center">
 # The entire main menubar is placed into messenger-menubar.inc.xhtml, so that it
 # can be shared with other top level windows.
 #include messenger-menubar.inc.xhtml
         </toolbaritem>
+#include messenger-titlebar-items.inc.xhtml
       </toolbar>
+      <html:unified-toolbar></html:unified-toolbar>
 
       <toolbar id="tabs-toolbar" class="chromeclass-toolbar">
         <tabs is="tabmail-tabs" id="tabmail-tabs"
               flex="1"
               align="end"
               setfocus="false"
               alltabsbutton="alltabs-button"
               collapsetoolbar="tabs-toolbar"
diff --git a/mail/themes/shared/mail/messenger.css b/mail/themes/shared/mail/messenger.css
--- a/mail/themes/shared/mail/messenger.css
+++ b/mail/themes/shared/mail/messenger.css
@@ -62,16 +62,21 @@ hr {
 
 #toolbar-menubar[autohide="true"][inactive="true"]:not([customizing="true"]) {
   min-height: 0 !important;
   height: 0 !important;
   padding: 0 !important;
   appearance: none !important;
 }
 
+/* Hide the minimise/fullscreen/close buttons when the menubar is visible */
+#toolbar-menubar:not([autohide=true]) ~ unified-toolbar .titlebar-buttonbox-container {
+  visibility: collapse;
+}
+
 #titlebar-spacer {
   pointer-events: none;
 }
 
 #navigation-toolbox:-moz-lwtheme,
 #ab-toolbox:-moz-lwtheme,
 #compose-toolbox:-moz-lwtheme {
   background-color: var(--lwt-accent-color);
